libraries/c_sdk/standard/mqtt_v2/CMakeLists.txt

```diff
@@ -23,10 +23,8 @@ afr_module_sources(
 
 afr_module_include_dirs(
     ${AFR_CURRENT_MODULE}
-    PUBLIC
+    INTERFACE
```

Comment:
Any reason you had for changing these to interface? I think they were already good as is. Usually, you would only be using INTERFACE for tests. If you would like more insight into the differences among PUBLIC vs PRIVATE vs INTERFACE, see [here](https://cmake.org/pipermail/cmake/2016-May/063400.html):
>    - When A links in B as *PRIVATE*, it is saying that A uses B in its
>    implementation, but B is not used in any part of A's public API. Any code
>    that makes calls into A would not need to refer directly to anything from
>    B. An example of this could be a networking library A which can be built to
>    use one of a number of different SSL libraries internally (which B
>    represents). A presents a unified interface for client code which does not
>    reference any of the internal SSL data structures or functions. Client code
>    would have no idea what SSL implementation (B) is being used by A, nor does
>    that client code need to care.
>    - When A links in B as *INTERFACE*, it is saying that A does not use B
>    in its implementation, but B is used in A's public API. Code that calls
>    into A may need to refer to things from B in order to make such calls. One
>    example of this is an interface library which simply forwards calls along
>    to another library but doesn't actually reference the objects on the way
>    through other than by a pointer or reference. Another example is where A is
>    defined in CMake as an interface library, meaning it has no actual
>    implementation itself, it is effectively just a collection of other
>    libraries (I'm probably over-simplifying here, but you get the picture).
>    - When A links in B as *PUBLIC*, it is essentially a combination of
>    PRIVATE and INTERFACE. It says that A uses B in its implementation and B is
>    also used in A's public API.

Reply:
Yeah, I am using managed MQTT API as INTERFACE in my shim. This is the reason I changed it to interface.

Reply:
I think the reason for INTERFACE to avoid statically compile the MQTT library. mqtt_config.h is required for the library to be statically compiled. mqtt_config.h is provided by the application of the library and hence it cannot be statically compiled as a library.